src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    private final SessionFactory factory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
```

Comment:
```suggestion
        try (Session session = factory.openSession()) {
```
and replace in other methods.